test/UID2.Client.Test/BidstreamClientTests.cs

```diff
@@ -81,6 +81,26 @@ private void SmokeTestForBidstream(IdentityScope identityScope, TokenVersion tok
             DecryptAndAssertSuccess(advertisingToken, tokenVersion);
         }
 
+        [Theory]
+        [InlineData(IdentityScope.UID2)]
+        [InlineData(IdentityScope.EUID)]
+        private void CanDecryptV4TokenEncodedAsBase64(IdentityScope identityScope)
+        {
+            Refresh(KeyBidstreamResponse(new[] { MASTER_KEY, SITE_KEY }, identityScope));
+
+            var now = DateTime.UtcNow;
+            var advertisingTokenBase64Url = AdvertisingTokenBuilder.Builder().WithVersion(TokenVersion.V4).WithScope(identityScope).WithEstablished(now.AddMonths(-4)).WithGenerated(now.AddDays(-1)).WithExpiry(now.AddDays(2)).Build();
+
+            var tokenAsBinary = UID2Base64UrlCoder.Decode(advertisingTokenBase64Url);
+            var advertisingTokenBase64 = Convert.ToBase64String(tokenAsBinary);
+            Assert.True(advertisingTokenBase64.Contains("="));
+            Assert.True(advertisingTokenBase64.Contains("/"));
```

Comment:
there is a slim chance that there may not be  / or + char? is it better to assert that ```advertisingTokenBase64Url``` must contain - and _ char first (if not, generate a new token value again) and then we are assured that we are testing the Bae64url->base64 conversion with these asserts?

Reply:
Added some code to regenerate in a loop until it finds all these base64 chars. Tested a few times, but in practice I've never seen it generate a token that doesn't have all of these chars.